openpype/pipeline/workfile/path_resolving.py

```diff
@@ -534,3 +538,36 @@ def create_workdir_extra_folders(
         fullpath = os.path.join(workdir, subfolder)
         if not os.path.exists(fullpath):
             os.makedirs(fullpath)
+
+
+def get_last_workfile_representation(
+    project_name: str,
+    asset_name: str,
+    task_name: str,
+) -> dict:
+    """Get last published workfile representation.
+
+    Args:
+        project_name(str): Project name.
+        asset_name(str): Asset/Shot name.
+        task_name(str): Task name.
+
+    Returns:
+        dict: Last workfile representation.
+    """
+    return max(
+        filter(
+            lambda r: r["context"].get("version") is not None,
+            list(
+                get_representations(
+                    project_name,
+                    context_filters={
+                        "asset": asset_name,
+                        "family": "workfile",
+                        "task": {"name": task_name},
+                    },
+                )
+            ),
+        ),
+        key=lambda r: r["context"]["version"],
+    )
```

Comment:
I feel like it's weird we're going straight to the representation to find the latest version. Shouldn't this somehow go through a call through `get_last_versions` or `get_last_version_by_subset_name` and then get the representation from there?

[AYON doesn't even seem to support it](https://github.com/pypeclub/OpenPype/blob/73e928efc99dbcd90b84d09cd1de0397ed3f633b/openpype/client/server/entities.py#L492)?

Reply:
Oof! Thanks for pointing at the incompatibility with Ayon. My implementation works the same as what was already in the `Copy Last Workfile` pre launch hook. So if doing it that way could cause an issue, it's a good occasion to change that!

Reply:
With MongoDB backend the query by context should still perform quite decent and is trivial to do so I assume that's why it was done that way at the time. Not sure how those concepts translate to the AYON backend. @martastain is the psycho usually having all the answers to that. 🥇